Specs/Core/loadWithXhrSpec.js

```diff
@@ -133,18 +143,269 @@ defineSuite([
             });
         });
 
-        xit('can support 2xx HTTP status (other than 200)', function(){
+        xit('can support 2xx HTTP status (other than 200)', function() {
             return loadWithXhr({
-                method: 'POST',
-                url: 'http://jsonplaceholder.typicode.com/posts',
-                data: {
-                    title: 'foo',
-                    body: 'bar',
-                    userId: 1
+                method : 'POST',
+                url : 'http://jsonplaceholder.typicode.com/posts',
+                data : {
+                    title : 'foo',
+                    body : 'bar',
+                    userId : 1
                 }
-            }).then(function(result){
+            }).then(function(result) {
                 expect(JSON.parse(result).id).toEqual(101);
             });
         });
     });
+
+    describe('URL loading using XHR', function() {
+        describe('returns a promise that rejects when the request', function() {
+            it('results in an HTTP status code greater than or equal to 300', function() {
+                return loadWithXhr({
+                    url : 'http://example.invalid'
+                }).then(function(resolvedValue) {
```

Comment:
Don't use `.then(resolveCallback, rejectCallback)`  Instead use `.then(resolveCallback).otherwise(rejectCallback).  The former does not catch all exceptions, while the latter will.